controllers/nodefeaturediscovery_controls.go

```diff
@@ -325,8 +325,16 @@ func DaemonSet(n NFD) (ResourceStatus, error) {
 			args = append(args, fmt.Sprintf("--instance=%s", n.ins.Spec.Instance))
 		}
 
-		if len(n.ins.Spec.Operand.ExtraLabelNS) != 0 {
-			args = append(args, fmt.Sprintf("--extra-label-ns=%s", strings.Join(n.ins.Spec.Operand.ExtraLabelNS, ",")))
+		if len(n.ins.Spec.ExtraLabelNs) != 0 {
+			args = append(args, fmt.Sprintf("--extra-label-ns=%s", strings.Join(n.ins.Spec.ExtraLabelNs, ",")))
+		}
+
+		if len(n.ins.Spec.ResourceLabels) != 0 {
+			args = append(args, fmt.Sprintf("--resource-labels=%s", strings.Join(n.ins.Spec.ResourceLabels, ",")))
+		}
```

Comment:
For consistency, I'd suggest to use `!= ""` in all of these

Reply:
Hi, @marquiz , thanks for the consistency check, but those two are lists, and is designed to be joined into a string as command args, IMHO, it may be not worth to join it into a string before checking the length, right?

Reply:
Ah, you're correct, sorry for the noise 😬 Forget about my comment 🙄